File: backend/src/app/schemas/location_tag_type.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

# Základní schéma pro LocationTagType
class LocationTagTypeBase(BaseModel):
    name: str
    # Přidáme další atributy jako barva, popis později, pokud budou potřeba

# Schéma pro vytváření LocationTagType (vstup)
class LocationTagTypeCreate(LocationTagTypeBase):
    pass

# Schéma pro aktualizaci LocationTagType (vstup)
class LocationTagTypeUpdate(BaseModel):
    name: Optional[str] = None

# Společné atributy uložené v DB
class LocationTagTypeInDBBase(LocationTagTypeBase):
    id: int
    world_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

# Finální schéma pro čtení LocationTagType (výstup)
class LocationTagType(LocationTagTypeInDBBase):
    pass 

File: frontend/src/types/locationTagType.ts
/**
 * Reprezentuje typ tagu pro lokaci (např. Město, Jeskyně, Obchod).
 */
export interface LocationTagType {
  id: number;
  world_id: number;
  name: string;
  created_at: string; // ISO date string
  updated_at: string; // ISO date string
  // Případně další vlastnosti jako color, description, pokud je přidáme
}

// Typ pro vytváření nového typu tagu
export interface LocationTagTypeCreate {
  name: string;
}

// Typ pro aktualizaci typu tagu
export interface LocationTagTypeUpdate {
  name?: string;
} 